### ISSUE TITLE
Not able to orderBy on one to many fields

### ISSUE THREAD

**Issue by ciprian1234** [prisma/prisma#21174]
## Problem

Hi everyone, I'm working on an API which relies on internationalization. I'm trying to order data by name alphabetically, but this use case seems to not be possible with Prisma.

Here its an oversimplified example:
![image](https://github.com/prisma/prisma/assets/17992599/afd8a1e4-9860-482a-8ded-b5d5ce02224b)

Here its what I'm trying to accomplish:
![image](https://github.com/prisma/prisma/assets/17992599/9012362a-7f8b-4c83-9709-fc0939956d0a)

**Comment by janpio** [comment]
Do I understand correctly that you want to read a list of things, their translation in _one_ specific language, and then sort by that translated value?

Or in code: Get a list of `someEntity`, with their `translations` included, then filter by a specific `locale` - and then sort everything by the translated name?

**Comment by ciprian1234** [comment]
Yes, I want to get a list of "someEntity" translated in a specific language, ordered by "someEntityName" (translated name).

**Comment by janpio** [comment]
Can you share your models and code as text please, so we don't have to fully type it manually? Thanks.

**Comment by ciprian1234** [comment]
yes sure:
```prisma
model Locale {
  locale            String  @id @db.VarChar(5)
  someEntitiesI18n  SomeEntity_I18n[]
}

model SomeEntity {
  id                String        @id @default(dbgenerated("uuid_generate_v4()")) @db.Uuid
  defaultName       String
  translations      SomeEntity_I18n[]
}

model SomeEntity_I18n {
  id                String        @id @default(dbgenerated("uuid_generate_v4()")) @db.Uuid
  locale            String        @db.VarChar(5)
  someEntityId      String        @db.Uuid
  someEntityName    String
  localeRef         Locale        @relation(fields: [locale], references: [locale])
  translatedItem    SomeEntity    @relation(fields: [someEntityId], references: [id])
  @@unique([locale, someEntityId])
}
```

```javascript
const data = await prisma.someEntity.findMany({
  take,
  skip,
  include: { translations: { where: { locale: 'en' } } }, // this will return only 1 item
  orderBy: { translations: { someEntityName: 'asc' } }, // NOT WORKING BECAUSE ITS ONE TO MANY RELATIONSHIP
});

**Comment by ciprian1234** [comment]
> @ciprian1234 I'm facing the same problem. Any updates on this ?

Although the queries that I wrote using Prisma are going to take some time and effort to rewrite using SQL, the only solution that I see right now in order to support sorting feature its to rewrite the queries using raw SQL.

**Comment by marvinruder** [comment]
> yes sure:
> 
> ```prisma
> model Locale {
>   locale            String  @id @db.VarChar(5)
>   someEntitiesI18n  SomeEntity_I18n[]
> }
> 
> model SomeEntity {
>   id                String        @id @default(dbgenerated("uuid_generate_v4()")) @db.Uuid
>   defaultName       String
>   translations      SomeEntity_I18n[]
> }
> 
> model SomeEntity_I18n {
>   id                String        @id @default(dbgenerated("uuid_generate_v4()")) @db.Uuid
>   locale            String        @db.VarChar(5)
>   someEntityId      String        @db.Uuid
>   someEntityName    String
>   localeRef         Locale        @relation(fields: [locale], references: [locale])
>   translatedItem    SomeEntity    @relation(fields: [someEntityId], references: [id])
>   @@unique([locale, someEntityId])
> }
> ```
> 
> ```js
> const data = await prisma.someEntity.findMany({
>   take,
>   skip,
>   include: { translations: { where: { locale: 'en' } } }, // this will return only 1 item
>   orderBy: { translations: { someEntityName: 'asc' } }, // NOT WORKING BECAUSE ITS ONE TO MANY RELATIONSHIP
> });
> ```

I had a similar issue and solved it by querying the other relation.

@ciprian1234 Does the following return the desired results for you?

```typescript
const data = (await prisma.locale.findUniqueOrThrow({
    where: {
        locale: "en",
    },
    select: {
        someEntitiesI18n: {
            skip,
            take,
            orderBy: {
                someEntityName: "asc",
            },
            include: {
                translatedItem: true,
            }
        }
    }
})).someEntitiesI18n.map((item) => ({
    ...item.translatedItem,
    translations: [item],
}));
```
